### ISSUE TITLE
error: unknown type name 'my_bool

### ISSUE THREAD

**Issue by imKota** [gearman/gearmand#165]
https://github.com/gearman/gearmand/blob/e2d76cf631414964decc2f22006a666e1c0e2424/libgearman-server/plugins/queue/mysql/queue.cc#L211

my_bool has been renamed to bool in MySQL 8.0.
https://dev.mysql.com/doc/refman/8.0/en/c-api-prepared-statement-data-structures.html

**Comment by esabol** [comment]
That doesn't sound like it has anything to do with macOS?

**Comment by imKota** [comment]
@esabol yeah, sorry))

**Comment by esabol** [comment]
What about MariaDB compatibility? Do they still use my_bool or have they switched to bool as well?

**Comment by esabol** [comment]
https://mariadb.com/kb/en/library/mysql_bind/ would seem to indicate that MariaDB still uses `my_bool`?

We'll probably need `configure` to determine whether to use `my_bool` or `bool` and set an appropriate `#define`... Something like the following wherever `my_bool` is currently used:

```C
 #ifndef HAVE_TYPE_MY_BOOL
 #include <stdbool.h>
 typedef bool my_bool;
 #endif
```

But how do we go about getting `configure` to test if `mysql.h` has `my_bool` or not?

**Comment by esabol** [comment]
I've also seen the following used:
```C
#if !defined(MARIADB_BASE_VERSION) && !defined(MARIADB_VERSION_ID) && \
  MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
#include <stdbool.h>
typedef bool my_bool;
#endif
```

See https://jira.mariadb.org/browse/MDEV-13773 and https://bugs.mysql.com/?id=87337

**Comment by p-alik** [comment]
fixed by @esabol in #212
